From 8bda02007c3f4527423f4131819ec397e854cc75 Mon Sep 17 00:00:00 2001
From: Davidlohr Bueso <dave@stgolabs.net>
Date: Mon, 14 May 2018 15:11:21 -0700
Subject: [PATCH] kvm: Fix nopvspin static branch init usage
Patch-mainline: never, SUSE specific
References: bsc#1056427

This fixes a warning in which the

smp_prepare_boot_cpu() ->
   native_smp_prepare_boot_cpu() ->
      native_pv_lock_init() ->
          if (!static_branch_likely(&virt_spin_lock_key))

smp_prepare_boot_cpu() is called from start_kernel() before
jump_label_init() happens, so this might potentially lead to all
kinds of  havoc. Move the function higher up outside of
smp_prepare_boot_cpu().

Signed-off-by: Davidlohr Bueso <dbueso@suse.de>

---
 arch/x86/kernel/smpboot.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index e53d700f409a..f13bbb604853 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -1216,6 +1216,8 @@ void __init native_smp_prepare_cpus(unsigned int max_cpus)
 	pr_info("CPU%d: ", 0);
 	print_cpu_info(&cpu_data(0));
 
+	native_pv_lock_init();
+
 	uv_system_init();
 
 	set_mtrr_aps_delayed_init();
@@ -1243,7 +1245,6 @@ void __init native_smp_prepare_boot_cpu(void)
 	/* already set me in cpu_online_mask in boot_cpu_init() */
 	cpumask_set_cpu(me, cpu_callout_mask);
 	cpu_set_state_online(me);
-	native_pv_lock_init();
 }
 
 void __init native_smp_cpus_done(unsigned int max_cpus)
-- 
2.13.6

